ui/src/common/state.ts

```diff
@@ -260,6 +260,7 @@ export interface TrackGroupState {
   parentGroup?: string; // Parent group id, if a subgroup.
   subgroups: string[]; // Child group ids.
   tracks: string[];  // Child track ids.
+  sortOrder: string[]; // Order to show children in the timeline
```

Comment:
This array is initially empty? So that if at any time it is empty, then the implied sort order is the groups in the order they appear in the `subgroups` property and tracks in the order they appear in the `tracks` property?

And what about the summary track of the group? It doesn't make sense to sort it anywhere but position zero, so how does that work? Does it matter? Or do we take steps to prevent changing its sort order?

What happens if the number of elements in the `sortOrder` is different to the number of elements in `subgroups` and `tracks` combined (possibly minus one for the summary track)? Does it matter?

The answers to these questions aren't clear to me from reading the code, and I think they should be.

Reply:
sortOrder is updated whenever a track or track group is added to its parent group.
```
assertExists(state.trackGroups[track.trackGroup]).tracks.push(id);
 assertExists(state.trackGroups[track.trackGroup]).sortOrder.push(id);
```
The same thing happens on removals
```
remove(parent.subgroups, groupId);
remove(parent.sortOrder, groupId);
```
If for any reason a track or track group child is not in its parent's sortOrder then it won't be drawn, but that shouldn't happen as long as we're using Perfetto Actions.

Also, regardless of whether or not the summary track gets into the sortOrder it is not being drawn as a normal child
```
for (const id of group.sortOrder) {
          if (group.tracks.includes(id) && id !== group.tracks[0]) {
```